fedimint-server/src/consensus/mod.rs

```diff
@@ -547,6 +549,31 @@ impl FedimintConsensus {
         }
     }
 
+    /// Adds any new shutdown signals to the set of signaling peers
+    async fn process_shutdown_signals(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        shutdown_signals: &[(PeerId, ShutdownReason)],
+    ) {
+        if !shutdown_signals.is_empty() {
+            let maybe_exists = dbtx.get_value(&ShutdownSignalKey).await;
+            let mut peers = maybe_exists.unwrap_or_default();
+            peers.extend(shutdown_signals.iter().map(|(peer, _)| peer));
+            dbtx.insert_entry(&ShutdownSignalKey, &peers).await;
```

Comment:
Personally not a big fan of editing DB entries, but saves a few DB key/prefix structs here. With DB transactions it's not a problem anymore.